tests/utils/state_locks_filter.py

```diff
@@ -0,0 +1,398 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import re
+import sys
+
+"""
+LockOpTrace: Represents a Lock Operation trace in the log file.
+It holds the lines for the trace and allows matching against a part of the trace.
+"""
+
+
+class LockOpTrace:
```

Comment:
doc comments go after type declaration

```suggestion
class LockOpTrace:
    """
    LockOpTrace represents a Lock Operation trace in the log file.
    It holds the lines for the trace and allows matching against a part of the trace.
    """
```